### PR TITLE
Bug 1205292 - Disallow new sync users. r=michielbdejong, yzen

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bded119ecc42fb9cfbb92d7da9555fe1b7dfaff6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=14eeeb3c444146dc3fd0cad0b51a8f27d38a4d56

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f557d4bab12b8a1514aeff130bd55ad04565e551

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cf75578d4ab316238d0c927f945b334bd3fc6a0e

File: apps/settings/locales/settings.en-US.properties [hunk 1]
```diff
@@ -843,6 +843,9 @@ fxsync-collections-passwords=Passwords
 fxsync-legal=Legal
 fxsync-tos=Terms of Service
 fxsync-privacy=Privacy Notice
+fxsync-error-invalid-account=This account is inactive
+fxsync-error-invalid-account-explanation=You need to login in Firefox or Firefox for Android and retry
```
Comment (yzen) [inline apps/settings/locales/settings.en-US.properties]: nit: You need to first login in Firefox or Firefox for Android and then try again.

File: apps/settings/test/unit/mock_dialog_service.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+define(function() {
```
Comment (yzen) [inline apps/settings/test/unit/mock_dialog_service.js]: Thanks for adding this mock. Would you be able to remove the existing ones (that recreate it in other tests?):
* test/unit/panels/addon_details/panel_test.js
* test/unit/panels/call_fdn_list/panel_test.js
* test/unit/panels/call_fdn_settings/panel_test.js
* test/unit/panels/developer/developer_test.js
* test/unit/panels/full_developer_mode/panel_test.js
* test/unit/panels/screen_lock/screen_lock_test.js
* test/unit/panels/simpin/simpin_test.js
* test/unit/panels/simpin_dialog/simpin_dialog_test.js
* test/unit/panels/usb_storage/usb_transfer_test.js
No worries if you don't have cycles for that.

File: apps/settings/test/unit/panels/firefox_sync/panel_test.js [hunk 3]
```diff
@@ -287,67 +299,159 @@ suite('Firefox Sync panel >', () => {
...
+        assert.isNull(subject.elements[element.name]);
+      });
+    });
+
+    LOGGED_OUT_SCREEN_ELEMENTS.forEach(element => {
+      if (!element || !element.event) {
+        return;
+      }
+      test(element.name + ' ' + element.event + ' listener should be set',
+           () => {
```
Comment (yzen) [inline apps/settings/test/unit/panels/firefox_sync/panel_test.js]: nit: indentation

File: apps/settings/test/unit/panels/firefox_sync/panel_test.js [hunk 4]
```diff
@@ -287,67 +299,159 @@ suite('Firefox Sync panel >', () => {
...
-          var spy = this.sinon.spy(subject, element.listener);
-          listenerSpies.set(element.name, spy);
-        });
-        subject.onsyncchange({
-          state: state
-        });
+    suiteSetup(() => {
+      subject = suiteSetupCommon();
+      setSubjectSpiesAndStubs(subject);
+      LOGGED_OUT_SCREEN_ELEMENTS.forEach(element => {
```
Comment (yzen) [inline apps/settings/test/unit/panels/firefox_sync/panel_test.js]: Can the part that sets up and tears down listenerSpies be taken out as a function, there are a lot of repeating things here and below.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=597b53ea4f11cf49972044d1b2a2463f5f399b36

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f39324216cb21422cb96624cdda1ad5c54a4e160

File: apps/settings/locales/settings.en-US.properties (hunk 1 above)
Comment (michielbdejong) [inline apps/settings/locales/settings.en-US.properties]: I think the sentence can be improved a bit, I find the "login in" and "Firefox or Firefox" repetitions a bit hard to read. Maybe:

You need to activate Firefox Sync on Desktop or Android first and then try again.

Comment (yzen) [inline apps/settings/locales/settings.en-US.properties]: Yes, I like this.

File: apps/settings/test/unit/mock_dialog_service.js (hunk 2 above)
Comment (ferjm) [inline apps/settings/test/unit/mock_dialog_service.js]: I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1216411 for this

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b559cfce63eccb1b9f2ab05e274862d0472224a5

**Pull request merged**